modules/core/src/main/java/org/apache/ignite/internal/processors/rest/protocols/tcp/GridTcpRestParser.java

```diff
@@ -246,6 +246,45 @@ else if (msg instanceof GridRouterRequest) {
         }
     }
 
+    /**
+     * Parses redis protocol message.
+     *
+     * @param ses Session.
+     * @param buf Buffer containing not parsed bytes.
+     * @param state Current parser state.
+     * @return Parsed packet.s
+     * @throws IOException If packet cannot be parsed.
+     * @throws IgniteCheckedException If deserialization error occurred.
+     */
+    private GridClientMessage parseRedisPacket(GridNioSession ses, ByteBuffer buf, ParserState state)
```

Comment:
Wouldn't it be better to move this method to GridRedisProtocolParser.java class and renaming it to "readPacket"? This would also allow us to use "private" visibility level for the "validate.." set of the new methods